clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -649,10 +722,12 @@ private void handleListOffsetResponse(Map<TopicPartition, Long> timestampsToSear
         }
 
         // create the fetches
-        Map<Node, FetchRequest> requests = new HashMap<>();
+        Map<Node, FetchRequest.Builder> requests = new HashMap<>();
         for (Map.Entry<Node, LinkedHashMap<TopicPartition, FetchRequest.PartitionData>> entry : fetchable.entrySet()) {
             Node node = entry.getKey();
-            FetchRequest fetch = new FetchRequest(this.maxWaitMs, this.minBytes, this.maxBytes, entry.getValue());
+            FetchRequest.Builder fetch = new FetchRequest.Builder(
+                    this.maxWaitMs, this.minBytes, entry.getValue()).
```

Comment:
You are right that the general pattern is that constructor arguments are required, other things are optional.  Ideally we would have ListOffsetRequest#Builder(Map<TopicPartition, PartitionData>) and ListOffsetRequest#Builder(Map<TopicPartition, Long>), but we can't do that because of a Java limitation... both constructors would have the same type once type erasure has been performed, which is not allowed.